build_runner/lib/src/generate/options.dart

```diff
@@ -89,6 +89,8 @@ class BuildOptions {
 
     Logger.root.level = logLevel;
 
+    var logListener = Logger.root.onRecord.listen(environment.onLog);
```

Comment:
note that I moved this up so that I could log inside of `TargetGraph.forPackageGraph`, figured I might as well submit that change too though.